x-pack/plugin/old-lucene-versions/src/main/java/org/elasticsearch/xpack/lucene/bwc/codecs/LegacyAdaptingPerFieldPostingsFormat.java

```diff
@@ -52,12 +48,11 @@
  * latter only supports Lucene 7 and above (as it was shipped with backwards-codecs of Lucene 9 that
  * only has support for N-2).
  *
- * This class can probably be removed once we are on Lucene 10 and Lucene50PostingsFormat is no longer
+ * This class can be removed once Elasticsearch gets upgraded to Lucene 11 and Lucene50PostingsFormat is no longer
  * shipped as part of bwc jars.
- *
- * Swapping out formats can be done via the {@link #getPostingsFormat(String) method}.
  */
-public abstract class LegacyAdaptingPerFieldPostingsFormat extends PostingsFormat {
+@UpdateForV10(owner = UpdateForV10.Owner.SEARCH_FOUNDATIONS)
+public final class LegacyAdaptingPerFieldPostingsFormat extends PostingsFormat {
```

Comment:
this no longer needs to be abstract and subclassed, because the two only subclasses we had did exactly the same thing in their getPostingsFormat method: 

```
        if (formatName.equals("Lucene50")) {
            return new BWCLucene50PostingsFormat();
        } else {
            return new BWCCodec.EmptyPostingsFormat();
        }     
```

This logic is now incorporated in this class.